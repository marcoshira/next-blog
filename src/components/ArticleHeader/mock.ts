import { PostMetadata } from '../../shared-types/postAttributes';

export default {
  id: '1',
  attributes: {
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    slug: 'teste-1',
    excerpt:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse auctor fermentum nisl, ut pharetra leo lacinia et. Etiam nec urna sed orci bibendum sodales.',
    allowComments: true,
    createdAt: '2022-03-25T18:18:34.944Z',
    updatedAt: '2022-03-27T20:41:54.493Z',
    tags: {
      data: [
        {
          id: '1',
          attributes: {
            displayName: 'UseContext',
            slug: 'use-context',
          },
        },
        {
          id: '3',
          attributes: {
            displayName: 'UseState',
            slug: 'use-state',
          },
        },
      ],
    },
    categories: {
      data: [
        {
          id: '1',
          attributes: {
            displayName: 'JavaScript',
            slug: 'java-script',
          },
        },
        {
          id: '4',
          attributes: {
            displayName: 'React',
            slug: 'react',
          },
        },
      ],
    },
    author: {
      data: {
        id: '4',
        attributes: {
          displayName: 'Júlia Temari',
          slug: 'julia-temari',
        },
      },
    },
    cover: {
      data: {
        id: '14',
        attributes: {
          url: 'https://res.cloudinary.com/dr5ot2zqr/image/upload/v1648156169/1920x1080_6aaecb929e.jpg',
          alternativeText: '1920x1080 1',
        },
      },
    },
    content:
      '<p>We’ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.</p><figure class="table"><table><thead><tr><th>Title 1</th><th>Title 2</th><th>Title 3</th><th>Title 4</th><th>Title 5</th></tr></thead><tbody><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr></tbody></table></figure><blockquote><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla,</span></p></blockquote><p>&nbsp;</p><figure class="image"><img src="https://source.unsplash.com/random/800x800"></figure><p>&nbsp;</p><figure class="image image-style-side"><img src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg" alt="imagem 9"></figure><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus. Imagem ao lado:&nbsp;</span>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla, </span>Teste de imagem centralizada</p><p>&nbsp;</p><figure class="image"><img src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg" alt="imagem 9"></figure><p><a href="https://google.com.br">Teste de link</a>&nbsp;</p><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);"><i>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla,</i></span></p><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);"><strong>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet.</strong></span></p><p><code><strong>Teste de código asdasd asad</strong></code></p><pre><code class="language-plaintext">import { PostMetadata } from \'../../shared-types/postAttributes\';\nimport { ArticleHeader } from \'../ArticleHeader\';\nimport { HtmlContent } from \'../HtmlContent\';\nimport * as Styled from \'./styles\';\n\nexport type PostProps = PostMetadata;\n\nexport const Post = ({ attributes }: PostProps) =&gt; {\n  return (\n    &lt;Styled.Wrapper&gt;\n      &lt;ArticleHeader {...attributes} /&gt;\n      &lt;Styled.Content&gt;\n        &lt;HtmlContent html={attributes.content} /&gt;\n      &lt;/Styled.Content&gt;\n    &lt;/Styled.Wrapper&gt;\n  );\n};</code></pre><p><strong>Teste de vídeo</strong></p><figure class="media"><oembed url="https://www.youtube.com/watch?v=heRRKDQepy8"></oembed></figure>',
  },
} as PostMetadata;
