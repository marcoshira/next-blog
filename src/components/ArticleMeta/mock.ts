import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-03-25T18:28:16.052Z',
  updatedAt: '2022-03-28T18:29:04.677Z',
  tags: {
    data: [
      {
        id: '2',
        attributes: {
          displayName: 'UseRef',
          slug: 'use-ref',
        },
      },
      {
        id: '4',
        attributes: {
          displayName: 'Postgres',
          slug: 'postgres',
        },
      },
    ],
  },
  categories: {
    data: [
      {
        id: '5',
        attributes: {
          displayName: 'Django',
          slug: 'django',
        },
      },
    ],
  },
  author: {
    data: {
      id: '2',
      attributes: {
        displayName: 'Lu√≠sa Santiago',
        slug: 'luisa-santiago',
      },
    },
  },
} as ArticleMetaProps;
