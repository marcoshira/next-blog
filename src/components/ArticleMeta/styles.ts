import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.mediumGray};
    font-style: italic;
    a {
      color: ${theme.colors.crimson};
      text-decoration: none;
      transition: all 300ms ease-in-out;
      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
