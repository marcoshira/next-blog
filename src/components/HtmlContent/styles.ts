import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: calc(${theme.font.sizes.small} + 0.2rem);
    line-height: 1.5;
    p {
      margin: ${theme.spacings.small} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.crimson};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }
    a:hover {
      filter: brightness(50%);
    }
    code {
      font-family: monospace;
      color: ${theme.colors.crimson};
      font-size: ${theme.font.sizes.small};
      background: ${theme.colors.grey};
      padding: 0.2rem;
      margin: 0.2rem;
    }
    pre {
      background: ${theme.colors.black};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.small} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
    pre code {
      color: inherit;
      background: inherit;
    }
    img {
      max-width: 100%;
    }
    .image {
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }
    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }
    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacings.medium} 0;
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.grey};
    }
    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }
    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.small} 0;
    }
    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.grey};
    }
    blockquote {
      border-left: 0.5rem solid ${theme.colors.crimson};
      color: ${theme.colors.mediumGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;
      .image-style-side {
        max-width: 100%;
        float: none;
        margin: 0;
      }
    }
  `}
`;
