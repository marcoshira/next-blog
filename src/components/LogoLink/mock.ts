import { LogoLinkProps } from '.';

export default {
  text: 'LogoLink',
  srcImg: 'https://source.unsplash.com/random/800x800',
  link: 'https://www.google.com.br/',
  newTab: true,
} as LogoLinkProps;
