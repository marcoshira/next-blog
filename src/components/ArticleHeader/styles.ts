import styled, { css } from 'styled-components';
import { Wrapper as ArticleMeta } from '../ArticleMeta/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    > h1 {
      margin-bottom: ${theme.spacings.medium};
      max-width: 80rem;
    }

    > p {
      margin-top: 0;
      max-width: 80rem;
    }

    > img {
      width: 100%;
      max-width: 80rem;
    }

    & ${ArticleMeta} {
      max-width: 80rem;
    }
  `}
`;
