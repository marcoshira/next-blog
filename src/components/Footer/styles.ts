import styled, { css } from 'styled-components';
import { Container as HtmlContent } from '../HtmlContent/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.2rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${HtmlContent} {
      font-size: ${theme.font.sizes.small};

      > p > a {
        color: ${theme.colors.black};
      }
    }
  `}
`;
