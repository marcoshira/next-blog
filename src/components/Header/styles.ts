import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

type DisplayText = {
  showText?: boolean;
};

const displayText = () => css`
  display: flex;
`;

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    align-content: center;
    justify-content: center;
    width: 100%;

    div {
      margin: ${theme.spacings.xlarge} ${theme.spacings.medium};
    }
  `}
`;

export const Container = styled.div<DisplayText>`
  ${({ theme, showText }) => css`
    display: none;
    ${showText && displayText()}
    flex-direction: column;
    max-width: 70rem;

    & ${Heading} {
      margin-top: 0;
      margin-bottom: ${theme.spacings.xsmall};
    }
  `}
`;
