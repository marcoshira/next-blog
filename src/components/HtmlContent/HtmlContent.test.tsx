import { screen } from '@testing-library/react';
import { HtmlContent } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';
import mock from '../ArticleHeader/mock';

const content = mock.attributes.content;

describe('<HtmlContent />', () => {
  it('should render with medium font', () => {
    renderTheme(<HtmlContent html="Children" />);
    expect(screen.getByText('Children')).toHaveStyle({
      'font-size': `calc(${theme.font.sizes.small} + 0.2rem)`,
    });
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<HtmlContent html={content} />);
    expect(container).toMatchSnapshot();
  });
});
