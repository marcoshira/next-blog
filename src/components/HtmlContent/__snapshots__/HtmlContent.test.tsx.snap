// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlContent /> should match snapshot 1`] = `
.c0 {
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c0 p {
  margin: 1.6rem 0;
}

.c0 a,
.c0 a:visited,
.c0 a:link {
  color: crimson;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c0 code {
  font-family: monospace;
  color: crimson;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c0 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 1.6rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c0 pre code {
  color: inherit;
  background: inherit;
}

.c0 img {
  max-width: 100%;
}

.c0 .image {
  line-height: 0;
  margin: 2.4rem 0;
}

.c0 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c0 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c0 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0 ul,
.c0 ol {
  margin: 2.4rem 4.0rem;
}

.c0 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c0 table {
  width: 100%;
  border-collapse: collapse;
  margin: 1.6rem 0;
}

.c0 table td,
.c0 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c0 blockquote {
  border-left: 0.5rem solid crimson;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

@media (max-width:768px) {
  .c0 {
    font-size: 2rem;
  }

  .c0 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <p>
      We’ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.
    </p>
    <figure
      class="table"
    >
      <table>
        <thead>
          <tr>
            <th>
              Title 1
            </th>
            <th>
              Title 2
            </th>
            <th>
              Title 3
            </th>
            <th>
              Title 4
            </th>
            <th>
              Title 5
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Content 1
            </td>
            <td>
              Content 2
            </td>
            <td>
              Content 3
            </td>
            <td>
              Content 3
            </td>
            <td>
              Content 3
            </td>
          </tr>
          <tr>
            <td>
              Content 1
            </td>
            <td>
              Content 2
            </td>
            <td>
              Content 3
            </td>
            <td>
              Content 3
            </td>
            <td>
              Content 3
            </td>
          </tr>
          <tr>
            <td>
              Content 1
            </td>
            <td>
              Content 2
            </td>
            <td>
              Content 3
            </td>
            <td>
              Content 3
            </td>
            <td>
              Content 3
            </td>
          </tr>
        </tbody>
      </table>
    </figure>
    <blockquote>
      <p>
        <span
          style="background-color:rgb(255,255,255);color:rgb(0,0,0);"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla,
        </span>
      </p>
    </blockquote>
    <p>
       
    </p>
    <figure
      class="image"
    >
      <img
        src="https://source.unsplash.com/random/800x800"
      />
    </figure>
    <p>
       
    </p>
    <figure
      class="image image-style-side"
    >
      <img
        alt="imagem 9"
        src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg"
      />
    </figure>
    <p>
      <span
        style="background-color:rgb(255,255,255);color:rgb(0,0,0);"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus. Imagem ao lado: 
      </span>
       
    </p>
    <p>
       
    </p>
    <p>
       
    </p>
    <p>
      <span
        style="background-color:rgb(255,255,255);color:rgb(0,0,0);"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla, 
      </span>
      Teste de imagem centralizada
    </p>
    <p>
       
    </p>
    <figure
      class="image"
    >
      <img
        alt="imagem 9"
        src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg"
      />
    </figure>
    <p>
      <a
        href="https://google.com.br"
      >
        Teste de link
      </a>
       
    </p>
    <p>
      <span
        style="background-color:rgb(255,255,255);color:rgb(0,0,0);"
      >
        <i>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla,
        </i>
      </span>
    </p>
    <p>
      <span
        style="background-color:rgb(255,255,255);color:rgb(0,0,0);"
      >
        <strong>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet.
        </strong>
      </span>
    </p>
    <p>
      <code>
        <strong>
          Teste de código asdasd asad
        </strong>
      </code>
    </p>
    <pre>
      <code
        class="language-plaintext"
      >
        import { PostMetadata } from '../../shared-types/postAttributes';
import { ArticleHeader } from '../ArticleHeader';
import { HtmlContent } from '../HtmlContent';
import * as Styled from './styles';

export type PostProps = PostMetadata;

export const Post = ({ attributes }: PostProps) =&gt; {
  return (
    &lt;Styled.Wrapper&gt;
      &lt;ArticleHeader {...attributes} /&gt;
      &lt;Styled.Content&gt;
        &lt;HtmlContent html={attributes.content} /&gt;
      &lt;/Styled.Content&gt;
    &lt;/Styled.Wrapper&gt;
  );
};
      </code>
    </pre>
    <p>
      <strong>
        Teste de vídeo
      </strong>
    </p>
    <iframe
      height="315"
      src="https://www.youtube.com/embed/heRRKDQepy8"
      width="560"
    >
      &lt;/oembed&gt;&lt;/figure" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen&gt;
    </iframe>
  </div>
</div>
`;
